Controllers/ProductController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using EFCoreServices.Models;
+using EFCoreServices.Repository;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace EFCoreServices.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        IProductRepository productRepository;
+        public ProductController(ProductRepository _productRepository)
+        {
+            productRepository = _productRepository;
+        }
+
+        [HttpGet]
+        [Route("GetProductsList")]
+        public async Task<IActionResult> GetProductsList()
+        {
+            try
+            {
+                var products = await productRepository.GetProductsList();
+                if(products == null)
+                {
+                    return NotFound();
+                }
+                return Ok(products);
+            }
+            catch(Exception)
+            {
+                return BadRequest();
+            }
+        }
+
+        [HttpGet]
+        [Route("GetProductByID")]
+        public async Task<IActionResult> GetProductByID(int? prodId)
+        {
+            try
+            {
+                var product = await productRepository.GetProductByID(prodId);
+                if(product == null)
+                {
+                    return NotFound();
+                }
+                return Ok(product);
+            }
+            catch(Exception)
+            {
+                return BadRequest();
+            }
+        }
+
+        [HttpPost]
+        [Route("AddProduct")]
+        public async Task<IActionResult> AddProduct(Product prod)
+        {
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    var product_Id = await productRepository.AddProduct(prod);
+                    if (product_Id > 0)
+                    {
+                        return Ok(prod);
+                    }
+                    return NotFound();
+                }
+                catch (Exception)
+                {
+                    return BadRequest();
+                }
+            }
+            return BadRequest();
```

Comment:
you need to ensure you pass on as much info to the user. 
 return BadRequest(ModelState.Dictionary); I think should have the errors